src/librustc_metadata/encoder.rs

```diff
@@ -1235,10 +1235,14 @@ impl<'a, 'b: 'a, 'tcx: 'b> IsolatedEncoder<'a, 'b, 'tcx> {
                     self.encode_optimized_mir(def_id)
                 }
                 hir::ItemConst(..) => self.encode_optimized_mir(def_id),
-                hir::ItemFn(_, _, constness, _, ref generics, _) => {
-                    let has_tps = generics.ty_params().next().is_some();
+                hir::ItemFn(_, _, constness, ..) => {
+                    let generics = tcx.generics_of(def_id);
+                    let has_types = generics.params.iter().any(|param| match param.kind {
+                        ty::GenericParamDefKind::Type { .. } => true,
+                        _ => false,
+                    });
```

Comment:
Again, isn't there a method on `ty::Generics` for this check?

Reply:
There's a way to count the number of types, but that won't return early. I could add a method if it's frequent enough.

Reply:
Again, I think it's `requires_monomorphization`.